invenio_records_resources/services/errors.py

```diff
@@ -55,3 +55,13 @@ def __init__(self, vocabulary_id):
         """Initialise error."""
         self.vocabulary_id = vocabulary_id
         super().__init__(_("Facet {vocab} not found.").format(vocab=vocabulary_id))
+
+
+class FileKeyNotFoundError(Exception):
+    """Error denoting that a record doesn't have a certain file."""
+
+    def __init__(self, recid, file_key):
+        """Constructor."""
+        super().__init__(f"Record '{recid}' has no file '{file_key}'.")
```

Comment:
This is missing formatting and translation?

```suggestion
        super().__init__(_("Record {recid} has no file {file_key}.").format(recid=recid, file_key=file_key))
```